### PR TITLE
Feat/Permission Template -- Feature Complete

### PR DESCRIPTION
**Pull request by GilTeraSky**
1. **What this PR does / why we need it**:
    This PR includes the complete implementation of a Permission Template.
2. **Which issue(s) this PR fixes**
3. **Additional information**
    It was decided to go with a data source because there is no real state for a permission template as it can be generated for a 
    non-existing credentials.
    
    The following issues were addressed as follows:
    a. When generating a permission template for EKS, the templateValue field might contains values which are not defined as 
        parameters in the CloudFormation template. The workaround which was applied is to decode the template, and remove the 
        undefined values. This values are being kept instead in another computed field "undefined_template_values".
    b. The CloudFormation template returned from the API defined only a "Message" field as output and which contains text and 
        the ARN. The workaround should be implemented in the Terraform code to split the message data and get the ARN only.

4. **Special notes for your reviewer**

### CODE REVIEW

**Comment by shobha2626** [comment]
@vmw-vjn having a guide section for permission template would be helpful?

**Comment by shobha2626** [comment]
Please squash the commits accordingly

File: internal/resources/permissiontemplate/tests/datasource_tf_configs.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+const (
+	dataProtectionPermissionTemplateDataSourceName = "data_protection_permissions"
+	eksPermissionTemplateDataSourceName            = "eks_permissions"
+)
+
+var (
+	dataProtectionPermissionTemplateDataSourceFullName = fmt.Sprintf("data.%s.%s", permissiontemplateres.ResourceName, dataProtectionPermissionTemplateDataSourceName) //nolint:unused
+	eksPermissionTemplateDataSourceFullName            = fmt.Sprintf("data.%s.%s", permissiontemplateres.ResourceName, eksPermissionTemplateDataSourceName)            //nolint:unused
```
[the remark this answers is not in the document]
Comment (GilTeraSky) [reply internal/resources/permissiontemplate/tests/datasource_tf_configs.go:21]: For some reason the linter recognizes these variables as unused but they are used in the test file.

**Comment by GilTeraSky** [comment]
> @vmw-vjn having a guide section for permission template would be helpful?

I included the entire Terraform config that utilizes the permission template, from my understanding a guide is needed when you need to create assets manually.
In this case it is very simple, you have a data source with certain fields to be used in another resource, therefore it is very straight forwards to implement, especially when you can see the entire Terraform.

**Review by vrabbi — DISMISSED** [review]

**Review by shobha2626 — APPROVED** [review]
LGTM

File: internal/client/permissiontemplate/permission_template_resource.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
```
Comment (ramya-bangera) [inline internal/client/permissiontemplate/permission_template_resource.go:2]: nit: update copyright to 2024

Comment (ankitsny) [reply internal/client/permissiontemplate/permission_template_resource.go]: done

**Review by ramya-bangera — APPROVED** [review]

**Review by shobha2626 — APPROVED** [review]

**Pull request merged**